python_modules/dagster/dagster/_core/execution/context/system.py

```diff
@@ -559,6 +559,16 @@ def __init__(
         self._is_external_input_asset_version_info_loaded = False
         self._data_version_cache: Dict[AssetKey, "DataVersion"] = {}
 
+        self._is_explicit_mode = False
+
+    # In explicit mode, no conversion is done on returned values and missing but expected outputs are not
+    # allowed.
+    def is_explicit_mode(self) -> bool:
+        return self._is_explicit_mode
+
+    def set_explicit_mode(self) -> None:
+        self._is_explicit_mode = True
+
```

Comment:
I think we need

1) a much more description name than this. Like "require fully typed event stream"
2) Make this hidden in outer context (with leading underscore, etc)
3) Allow a way to register customized error messages. The entire point is communicate more clear errors to users.

Reply:
Implemented (1) and (3).

For (2), the method is already not `@public` and is intended to be called outside the class, so I don't see why we would `_`-prefix it.

Reply:
Yeah I guess I don't believe a user or customer calling a method not prefixed with `_` but not marked `@public` will accept us breaking them with the explanation of "it is not marked public"

Reply:
This is currently how we delineate user-facing from non-user methods all over the codebase on exported classes, including `OpExecutionContext`. Methods that aren't `@public` don't show up in docs. See for example `OpExecutionContext.{has_events,consume_events,get_step_execution_context}`. There are also many methods on `DagsterInstance` like this.

I realize this system is not great with typeahead, but fixing that is a separate problem. It would be confusing to treat just this one method in this category differently.